app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/room/RoomOfflineDatabase.kt

```diff
@@ -1,17 +1,39 @@
 package com.github.HumanLearning2021.HumanLearningApp.room
 
+import android.content.Context
 import androidx.room.Database
+import androidx.room.Room
 import androidx.room.RoomDatabase
 import androidx.room.TypeConverters
 
 @TypeConverters(RoomTypeConverters::class)
 @Database(
-    entities = [RoomCategory::class, RoomDatasetWithoutCategories::class, RoomUser::class, RoomPicture::class, RoomUnlinkedRepresentativePicture::class, RoomDatasetCategoriesCrossRef::class],
+    entities = [RoomCategory::class, RoomDatasetWithoutCategories::class, RoomUser::class, RoomPicture::class, RoomUnlinkedRepresentativePicture::class,
+        RoomDatasetCategoriesCrossRef::class, RoomEmptyHLDatabase::class, RoomDatabaseDatasetsCrossRef::class, RoomDatabaseCategoriesCrossRef::class, RoomDatabasePicturesCrossRef::class],
     version = 1,
     exportSchema = false
 )
 abstract class RoomOfflineDatabase : RoomDatabase() {
     abstract fun categoryDao(): CategoryDao
     abstract fun datasetDao(): DatasetDao
     abstract fun userDao(): UserDao
+    abstract fun databaseDao(): DatabaseDao
+
+    companion object {
+        // Singleton prevents multiple instances of the database opening at the same time
+        @Volatile
+        private var INSTANCE: RoomOfflineDatabase? = null
+
+        fun getDatabase(context: Context): RoomOfflineDatabase {
+            return INSTANCE ?: synchronized(this) {
+                val instance = Room.databaseBuilder(
```

Comment:
race condition, but that should be fixed by just injecting

Reply:
I did not manage to get this to work. I have a hard time with injecting into non-natively supported classes by Hilt

Reply:
wdym non-natively supported classes?

Reply:
Nevermind, I had a wrong understand of how it works... I will try again

Reply:
done